app/models/user.rb

```diff
@@ -80,6 +81,10 @@ def self.from_merged_attrs(existing_user, new_user)
     User.new(attrs)
   end
 
+  def self.from_saml(saml_response)
+    User.new(SAML::DeSerializer.new(saml_response).user_attributes)
```

Comment:
I like the rename, but I don't think invoking `user_attributes` directly from the initializer will work.

Consider:
```ruby
class UserAttributes
  def initialize(saml_response)
    @saml_response = saml_response
    user_attributes
  end
  ...
end

# inside user.rb
# this returns an instance, e.g. <SAML::UserAttributes:0x00000003e384a8 @saml_response=#<OneLogin::RubySaml::Response>, @raw_loa="http://idmanagement.gov/ns/assurance/loa/3">
# but we want a hash, e.g. {:uuid => '1111222', :first_name => 'bill', ... }
SAML::UserSettings.new(saml_response)  
```